### PR TITLE
Add xunit project to PowerShell.sln and make it runable from within VisualStudio

### LINKED ISSUE #7253: XUnit test project not part of Visual Studio solution PowerShell.sln -> make XUnit runable from within VS test explorer
Because `xunit` tests are running now on Windows and not just Linux only, it would make sense to add it to `Powershell.sln` solution file.
However, this does not work out of the box and needs some time to figure out how to adapt the `csproj` for that (it does not work out of the box with the upgrade that VS suggests). Any pointers on that?
I have a working prototype in my branch [here](https://github.com/bergmeister/PowerShell/commit/d6e736274e98531d92d0d7bbeadd30804c28b13b), will open a PR

### PR DESCRIPTION
**Pull request by bergmeister**
## PR Summary

- Resolves #7253 by adding the XUnit test project to `PowerShell.sln` and tweaking it to make the  project load in VS (by adding the `ToolsVersion`) and making it runable in the VS test explorer by adding the required NuGet packages `xunit.runner.visualstudio`.
- Related #6977 by updating the `xunit` package from `2.4.0-beta.2.build4010` to `2.4.0` (RTM) that got recently released

image

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Review by iSazonov — APPROVED** [review]

File: PowerShell.sln [hunk 1]
```diff
@@ -159,6 +153,14 @@ Global
 		{73EA0BE6-C0C5-4B56-A5AA-DADA4C01D690}.Debug|Any CPU.Build.0 = Linux|Any CPU
 		{73EA0BE6-C0C5-4B56-A5AA-DADA4C01D690}.CodeCoverage|Any CPU.ActiveCfg = Linux|Any CPU
 		{73EA0BE6-C0C5-4B56-A5AA-DADA4C01D690}.CodeCoverage|Any CPU.Build.0 = Linux|Any CPU
+		{190BB016-1395-4944-853F-98F8A4277483}.CodeCoverage|Any CPU.ActiveCfg = Release|Any CPU
```
Comment (anmenaga) [inline PowerShell.sln]: Can you please explain why these changes to ConfigurationPlatforms are necessary?

**Comment by adityapatwardhan** [comment]
@bergmeister Please have a look at merge conflicts.

**Comment by bergmeister** [comment]
@adityapatwardhan I cannot take the upstream changes because PR #6926 invalidated the PowerShell solution, which even prevents the XUnit project now from loading. I will therefore wait until the solution has been fixed in master.

File: PowerShell.sln [hunk 2]
```diff
@@ -123,14 +125,6 @@ Global
...
 		{07BFD271-8992-4F34-9091-6CFC3E224A24}.CodeCoverage|Any CPU.ActiveCfg = CodeCoverage|Any CPU
 		{07BFD271-8992-4F34-9091-6CFC3E224A24}.CodeCoverage|Any CPU.Build.0 = CodeCoverage|Any CPU
-		{07BFD271-8992-4F34-9091-6CFC3E224A24}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{07BFD271-8992-4F34-9091-6CFC3E224A24}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{07BFD271-8992-4F34-9091-6CFC3E224A24}.Linux|Any CPU.ActiveCfg = Linux|Any CPU
-		{07BFD271-8992-4F34-9091-6CFC3E224A24}.Linux|Any CPU.Build.0 = Linux|Any CPU
-		{07BFD271-8992-4F34-9091-6CFC3E224A24}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{07BFD271-8992-4F34-9091-6CFC3E224A24}.Release|Any CPU.Build.0 = Release|Any CPU
-		{8F63D134-E413-4181-936D-D82F3F5F1D85}.CodeCoverage|Any CPU.ActiveCfg = CodeCoverage|Any CPU
-		{8F63D134-E413-4181-936D-D82F3F5F1D85}.CodeCoverage|Any CPU.Build.0 = CodeCoverage|Any CPU
```
Comment (daxian-dbw) [inline PowerShell.sln]: You removed these two entries that are associated with `8F63D134-E413-4181-936D-D82F3F5F1D85`, but left two entries associated with `07BFD271-8992-4F34-9091-6CFC3E224A24` above. Is that intentional?

Comment (iSazonov) [reply PowerShell.sln]: I believe it is better to rebase then try to resolve conflicts in such auto changed file.

Comment (bergmeister) [reply PowerShell.sln]: The conflict is already resolved and I resolved it by taking HEAD and adding the project again to the solution

Comment (daxian-dbw) [reply PowerShell.sln]: @bergmeister Please take a look at the highlighted entries in the screenshot below. It looks to me you shouldn't remove the two `8F63D134` entries.

image

Comment (bergmeister) [reply PowerShell.sln]: @daxian-dbw Sorry, thanks for reminding me to address your comment as well. Yes, I made 3 commits to address this:
- Remove the 2 redundant `07BFD271` lines
- Revert the removal of the 2 `8F63D134` lines
- Reorder lines so that VS does not automatically re-order them on save,

**Comment by bergmeister** [comment]
UPDATE: I addressed the comments about the solution file and also updated the XUnit Nuget package to its RTM version as it got published today.

**Comment by daxian-dbw** [comment]
> Shall I also update the xunit reference in test/hosting/hosting.tests.csproj from 2.3.1 to 2.4.0?

I'm fine with updating the xunit reference, as long as it doesn't break our xunit test run 😄

**Review by SteveL-MSFT — APPROVED** [review]

**Review by TravisEz13 — APPROVED** [review]

**Review by anmenaga — APPROVED** [review]

**Comment by anmenaga** [comment]
@adityapatwardhan This looks ready for merge.

**Comment by adityapatwardhan** [comment]
@bergmeister Thank you for your contribution!

File: test/hosting/hosting.tests.csproj [hunk 3]
```diff
@@ -15,7 +15,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="xunit" Version="2.3.1" />
+    <PackageReference Include="xunit" Version="2.4.0" />
     <!-- DotNetCliToolReference element specifies the CLI tool that the user wants to restore in the context of the project. -->
     <DotNetCliToolReference Include="dotnet-xunit" Version="2.3.1" />
```
Comment (iSazonov) [inline test/hosting/hosting.tests.csproj]: @bergmeister We skipped the old version :-)

Comment (bergmeister) [reply test/hosting/hosting.tests.csproj]: Did you mean `dotnet-xunit`? Version 2.4.0 of it is still in preview
https://www.nuget.org/packages/dotnet-xunit/2.4.0-beta.1.build3958

Comment (iSazonov) [reply test/hosting/hosting.tests.csproj]: Yes, we add it in csharp.tests.csproj in the PR.

Comment (bergmeister) [reply test/hosting/hosting.tests.csproj]: Ok, I see what you mean now. If it's not essential, then I'd still just wait for 2.4.0 of this package to become RTM.

Comment (iSazonov) [reply test/hosting/hosting.tests.csproj]: We can wait. My comment is informational only.

**Comment by powercode** [comment]
@bergmeister Really happy about this PR! Thx!

**Comment by bergmeister** [comment]
Thanks @powercode , I also plan to do something similar for the hosting tests. Thanks for your efforts about the ReSharper settings as well.
